### ISSUE TITLE
Error When Trying To Access Private Github Repo

### ISSUE THREAD

**Issue by SC7639** [atom/github#1054]
### Description

In the GitHub (preview) tab is displayed "An unknown error has occured"

After checking the developer tools Network tab. The below is the response from https://api.github.com/graphql

```
{"query":"query Pr {relay {...F4}} fragment F0 on User {url,id} fragment F1 on Bot {url,id} fragment F2 on Node {id,__typename} fragment F3 on PullRequest {url,number,title,state,createdAt,author {login,avatarUrl,__typename,...F0,...F1,...F2},repository {name,owner {login,id,__typename},id},reactionGroups {content,users {totalCount}},commits {totalCount},_labels3D9oIE:labels(first:100) {edges {node {name,color,id},cursor},pageInfo {hasNextPage,hasPreviousPage}},id} fragment F4 on Query {_repositoryeNnn7:repository(owner:\"thecrowngroup\",name:\"vin\") {_pullRequestsk0zh:pullRequests(first:30,headRefName:\"master\") {totalCount,edges {node {id,number,title,url,...F3},cursor},pageInfo {hasNextPage,hasPreviousPage}},id}}","variables":{}}
```

I can push and pull from/to the repo in atom
![image](https://user-images.githubusercontent.com/6049560/28586873-121eab98-716d-11e7-9a12-e7aeede12559.png)

### Steps to Reproduce

1. Open a private repo in atom
2. Open tab on the right using ctrl + shift + 8 to see the GitHub tab

**Expected behavior:**

I expected to see pull requests if there are any or something like the below screenshot 
![image](https://user-images.githubusercontent.com/6049560/28586746-b1f71912-716c-11e7-86db-08ee75867d43.png)

**Actual behavior:**

An error has occurred has occurred is displayed in the GitHub tab
![image](https://user-images.githubusercontent.com/6049560/28586611-4c3bace6-716c-11e7-89e2-be87921be807.png)

**Reproduces how often:**

This happens every time I open the GitHub tab for this repo

### Versions
atom --version output
``` 
Atom    : 1.18.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

apm --version
```
apm  1.18.1
npm  3.10.10
node 6.9.5 x64
python 2.7.13
git 2.13.3.windows.1
visual studio 2015
```

**Comment by daveysigma** [comment]
I get the same problem. I think the login I provided for the repo may have been wrong. but there doesn't seem to be a place to unset the GitHub login credentials

**Comment by HarveyThayer** [comment]
I fixed it by going to
github.com > settings > Authorized OAuth Apps > GitHub for Atom > grant permissions

**Comment by bronson** [comment]
I have this problem too, with a private repo owned by an organization.  They're a little paranoid, not inclined to change anything related to security.

This means I can't use the github package at all, for any repo.  😞 

I do hope this can be fixed.

**Comment by bronson** [comment]
Not true!  If I restart Atom then I seem to be able to use the github package on authorized repos.

That's a relief.

**Comment by fenkner** [comment]
It works for me with public repos, but I need to request permission from a private repo owner for it to work even though I have access to that repo every other way.  I tried restarting Atom, but had the same problem.  Thanks to the earlier post by @HarveyThayer, I realized I had to click a button to request Atom access to a private repo.

**Comment by kibernick** [comment]
Thank you folks, I didn't realise what was the problem with my Atom installation until I found this issue! "An unknown error occurred." does not suffice and would be nice to have a message/link about granting permissions on your Github settings.

**Comment by bronson** [comment]
I tried authorizing the github package for a different private organization.  Even though GitHub shows a note saying that an email requesting authorization was sent, they never received an email.

And, there doesn't appear to be any ability to cancel or re-send the email.  If that first one disappears, you're locked in that state forever.  :)

As others said here, telling the owners of the org to manually go into the oauth settings and enable the permission did work.

**Comment by rsese** [comment]
Are you still having trouble @SC7639?  If so and you were having trouble with a private organization repository, you may need to authorize the Atom for GitHub app [1] [2] as other folks have mentioned in the thread.

[1]: https://help.github.com/articles/approving-oauth-apps-for-your-organization/
[2]: https://help.github.com/articles/requesting-organization-approval-for-oauth-apps/

**Comment by trusktr** [comment]
FWIW, I've never been able to use the "GitHub" panel, as much as I'd love to see what it can do.

At the moment, I'm getting the message `Could not resolve to a Repository with the name 'insert-name-here'`.

<img width="404" alt="Screen Shot 2019-05-14 at 2 33 53 PM" src="https://user-images.githubusercontent.com/297678/57733929-8b20ef80-7655-11e9-8525-a02ce9d55269.png">

Looking forward to eventually seeing what this tab can do one day...

**Comment by ibizaman** [comment]
@trusktr I had a similar error using forge in Emacs. Anyway, it works for me if I set `github.user` and `github.email` in my git config. https://github.com/magit/forge/issues/192#issuecomment-528386826 gave me the idea.

**Comment by beetle442002** [comment]
> I fixed it by going to
> github.com > settings > Authorized OAuth Apps > GitHub for Atom > grant permissions

Thanks, 
And because this is still relevant:

github.com -> Your Profile icon -> Settings -> Applications -> Authorized OAuth Apps -> Atom GitHub package (the name not the three dots) -> grant permissions

**Comment by Jeniffen** [comment]
> FWIW, I've never been able to use the "GitHub" panel, as much as I'd love to see what it can do.
> 
> At the moment, I'm getting the message `Query errors reported. Could not resolve to a Repository with the name 'insert-name-here'`.
> 
> <img alt="Screen Shot 2019-05-14 at 2 33 53 PM" width="404" src="https://user-images.githubusercontent.com/297678/57733929-8b20ef80-7655-11e9-8525-a02ce9d55269.png">
> 
> Looking forward to eventually seeing what this tab can do one day...

I had the exact same issue - I was able to fix it for me by 

1. [adding an SSH key ](https://docs.github.com/en/github/authenticating-to-github/checking-for-existing-ssh-keys) that was generated on my machine to my GitHub 
2. (Re-)Cloning my the project I was working on using the SSH URL
3. enter the token you can generate from the 'GitHub for Atom' panel

**Comment by andrewfirth05** [comment]
> I fixed it by going to github.com > settings > Authorized OAuth Apps > GitHub for Atom > grant permissions

Thanks Harvey!! This works for me - needed to grant permission to that repo under the Organisation Access section. Refreshed the connection in Atom and it now works :)
